util/src/main/scala/org/ensime/util/Debouncer.scala

```diff
@@ -0,0 +1,118 @@
+// Copyright: 2010 - 2017 https://github.com/ensime/ensime-server/graphs
+// License: http://www.gnu.org/licenses/gpl-3.0.en.html
+package org.ensime.util
+
+import akka.actor.Scheduler
+import java.time.Clock
+import java.util.concurrent.atomic.{ AtomicBoolean, AtomicInteger }
+import org.slf4j.LoggerFactory
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+
+/**
+ * Generic debouncer implementation
```

Comment:
I'm fine with this not being akka code, just so long as we document why. I tend to like comments to explain unexpected things, and performance optimisations are almost always unexpected things.